lib/rubocop/cop/rspec/mixin/top_level_group.rb

```diff
@@ -9,7 +9,6 @@ module TopLevelGroup
 
         def on_new_investigation
           super
-          return unless root_node
```

Comment:
Without filtering by`RSpec` department:
```
rubocop --fail-level E --display-only-fail-level-offenses ../real-world-rspec
```

```
49799 files inspected, 713 offenses detected

15 errors occurred:
An error occurred while InternalAffairs/UselessMessageAssertion cop was inspecting /Users/pirj/source/real-world-rspec/chef/spec/unit/action_collection_spec.rb.
An error occurred while InternalAffairs/UselessMessageAssertion cop was inspecting /Users/pirj/source/real-world-rspec/chef/spec/unit/application/agent_spec.rb.
An error occurred while InternalAffairs/UselessMessageAssertion cop was inspecting /Users/pirj/source/real-world-rspec/chef/spec/unit/application/server_spec.rb.
An error occurred while InternalAffairs/UselessMessageAssertion cop was inspecting /Users/pirj/source/real-world-rspec/chef/spec/unit/resource_builder_spec.rb.
An error occurred while InternalAffairs/UselessMessageAssertion cop was inspecting /Users/pirj/source/real-world-rspec/loomio/spec/extras/queries/users_to_email_query_spec.rb.
An error occurred while Style/CaseLikeIf cop was inspecting /Users/pirj/source/real-world-rspec/puppet/lib/puppet/parser/scope.rb:734:4.
An error occurred while InternalAffairs/UselessMessageAssertion cop was inspecting /Users/pirj/source/real-world-rspec/puppet/spec/unit/face_spec.rb.
An error occurred while InternalAffairs/UselessMessageAssertion cop was inspecting /Users/pirj/source/real-world-rspec/puppet/spec/unit/pops/containment_spec.rb.
An error occurred while Metrics/ClassLength cop was inspecting /Users/pirj/source/real-world-rspec/rspec-core/lib/rspec/core/formatters/protocol.rb:19:6.
An error occurred while InternalAffairs/UselessMessageAssertion cop was inspecting /Users/pirj/source/real-world-rspec/rspec-core/spec/rspec/core/resources/a_spec.rb.
An error occurred while InternalAffairs/UselessMessageAssertion cop was inspecting /Users/pirj/source/real-world-rspec/rspec-core/spec/rspec/core/resources/acceptance/foo_spec.rb.
An error occurred while Lint/InterpolationCheck cop was inspecting /Users/pirj/source/real-world-rspec/rubocop/spec/rubocop/cop/lint/literal_as_condition_spec.rb:6:23.
An error occurred while InternalAffairs/UselessMessageAssertion cop was inspecting /Users/pirj/source/real-world-rspec/spree/frontend/spec/features/caching/products_spec.rb.
An error occurred while InternalAffairs/UselessMessageAssertion cop was inspecting /Users/pirj/source/real-world-rspec/spree/frontend/spec/features/caching/taxons_spec.rb.
An error occurred while InternalAffairs/UselessMessageAssertion cop was inspecting /Users/pirj/source/real-world-rspec/spree/frontend/spec/features/microdata_spec.rb.
Errors are usually caused by RuboCop bugs.
```

Reply:
Whoops.

If you can you narrow the reproduction steps a bit, then we’ll write a spec to catch it so I don’t remove that line again in 3 months :-)

Reply:
Those are all RuboCop core cops, we shouldn't care much @bquorning @Darhazer 

We can report it though.

Reply:
Oh, I see.

Reply:
I reasoned it would be save to remove this `nil` check, since the only thing happening after it is we loop over `top_level_groups`, which is based on `top_level_nodes(root_node)`. And `top_level_nodes` will return `[]` if `root_note.nil?`. Meaning that no work will be done.